envs/runners/adversarial_runner.py

```diff
@@ -549,14 +582,26 @@ def agent_rollout(self,
                                 truncated_obs = info['truncated_obs']
                                 agent.storage.insert_truncated_obs(truncated_obs, index=i)
 
-                        # If using PLR, sample next level
+                        # If using PLR, sample next level 
+                        # don't do this if using accel-paired
+                        # since we want antagonist to rollout on same levels as that of protagonist
                         if level_sampler and level_replay:
-                            level_seed = level_sampler.sample_replay_level()
-                            level = self.level_store.get_level(level_seed)
-                            obs_i = self.venv.reset_to_level(level, i)
-                            set_obs_at_index(obs, obs_i, i)
-                            next_level_seeds[i] = level_seed
-                            rollout_info['solved_idx'][i] = True
+                            if self.use_accel_paired:
+                                level_seed = self.current_level_seeds[i]
+                                level = self.level_store.get_level(level_seed)
+                                obs_i = self.venv.reset_to_level(level, i)
+                                set_obs_at_index(obs, obs_i, i)
+                                next_level_seeds[i] = level_seed
+                                rollout_info['solved_idx'][i] = True
+                                self.current_level_seeds[i] = level_seed
+                            else:
+                                level_seed = level_sampler.sample_replay_level()
+                                level = self.level_store.get_level(level_seed)
+                                obs_i = self.venv.reset_to_level(level, i)
+                                set_obs_at_index(obs, obs_i, i)
+                                next_level_seeds[i] = level_seed
+                                rollout_info['solved_idx'][i] = True
+                                # self.current_level_seeds[i] = level_seed
```

Comment:
This line was present in the old `paired` repository but was removed in this DCD repo. Let me know if this is needed here or not.

Reply:
Removing this line